### PR TITLE
Add AppStream info file and extend .desktop file

### LINKED ISSUE #1066: Add Appstream description
Currently keepassxc is not visible to the user when using a software center like ``gnome-software``. This would require keepassxc to provide an AppStream description and could help more casual users to easily find and install keepassxc in distributions like Ubuntu or Fedora.

## Expected Behavior
The enduser can install Keepassxc via the software center provided by the distribution, can take a look at screenshots and comment/rate the application.

## Current Behavior
Searching for Keepassxc in ``gnome-software`` returns no result regardless whether the distribution ships it or not.

## Possible Solution
Provide AppStream data for distributions like in the [quickstart guide](https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html) explained. Further more it would be possible for extensions to provide a description for their relation to keepassxc.

Since AppStream do contain a description of where to look for screenshots another question would be of where to put those. And if the screenshots should be the same for every version regardless of UI changes or if those screenshots should be updated on demand and only be visible for those newer versions.

### PR DESCRIPTION
**Pull request by phoerious**
## Description

This patch resolves #1066 by adding an AppStream info file. It also extends the .desktop file with additional key/value pairs and changes its file name to match the specification.

Required pre-merge checks were added to the release-tool.

## How has this been tested?

Merge checks have been tested to work properly.

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: share/linux/org.keepassxc.desktop [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+GenericName=Password Manager
 GenericName[de]=Passwortverwaltung
 GenericName[es]=Gestor de contraseñas
 GenericName[fr]=Gestionnaire de mot de passe
 GenericName[ru]=менеджер паролей
+Comment=Community-driven port of the Windows application “KeePass Password Safe”
 Exec=keepassxc %f
+TryExec=keepassxc
 Icon=keepassxc
+Categories=Utility;Security;Qt
```
Comment (TheZ3ro) [inline share/linux/org.keepassxc.desktop]: Categories is repeated 2 times

Comment (phoerious) [reply share/linux/org.keepassxc.desktop]: Fixed.

File: share/linux/org.keepassxc.desktop [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 GenericName[fr]=Gestionnaire de mot de passe
 GenericName[ru]=менеджер паролей
+Comment=Community-driven port of the Windows application “KeePass Password Safe”
 Exec=keepassxc %f
+TryExec=keepassxc
 Icon=keepassxc
 Terminal=false
 Type=Application
-Categories=Qt;Utility;
+Version=1.0
```
Comment (TheZ3ro) [inline share/linux/org.keepassxc.desktop]: Why version is set to 1.0 instead os 2.2.1 ?

Comment (phoerious) [reply share/linux/org.keepassxc.desktop]: It's the spec version, not the app version.

**Comment by droidmonkey** [comment]
i will test against the snap build this evening.

**Pull request merged**